iter/filter_test.go

```diff
@@ -52,3 +69,28 @@ func TestExclude(t *testing.T) {
 	assert.Equal(t, evens.Next().Unwrap(), 1)
 	assert.Equal(t, evens.Next().Unwrap(), 3)
 }
+
+func TestFilterMap(t *testing.T) {
+	selectEvenAndDouble := func(x int) option.Option[int] {
+		if x%2 > 0 {
+			return option.None[int]()
+		}
+
+		return option.Some(x * 2)
+	}
+
+	fltMap := iter.FilterMap[int](
+		iter.Lift([]int{1, 2, 3, 4, 5, 6}),
+		selectEvenAndDouble,
+	)
+	fltMapEmpty := iter.FilterMap[int](
```

Comment:
We should be testing only one behaviour in each test. Please separate this into two tests `TestFilterMap` and `TestFilterMapEmpty`.